istioctl/cmd/analyze.go

```diff
@@ -56,12 +62,19 @@ var (
 		diag.Warning: "\033[33m",   // yellow
 		diag.Error:   "\033[1;31m", // bold red
 	}
+
+	msgOutputFormats    = map[string]bool{LogOutput: true, JsonOutput: true, YamlOutput: true}
+	msgOutputFormatKeys []string
```

Comment:
I dislike how we're creating this structure multiple times. Once as a map for existence, once as a list for output, once as a switch for the actual logic. 

I'm not sure what the most elegant way to handle this is in Go is, though. :(

Brainstorming some ideas you are under no obligation whatsoever to consider:

* Instead of a switch, define the map value to be a function that takes in Messages and returns a string. Or alternately, have something like a `MessageWriter` interface type. Not sure that ends up being easier to read in the end even if it feels more Go-ey and robust. 
* Get rid of the map, just have the possible formats as a list. Either don't validate before analysis at all (and rely on that "default" case to catch errors), or just iterate through the list once to find a match. (Why u no have a "contains" method, Go?)
* Move msgOutputFormatKeys into the Analyze() function. There's no need for it to be a global var.

Reply:
I don't like it either :( I really want something akin to Python's dictionary functions here. This was the closest balance I could find to using the same format as proxyconfig.go which also takes an outputFormat argument, and being Go-ey about explicit loops and things.

I'll move msgOutputFormatKeys into Analyze, you're right that there's no need for it to be a global var. If you don't mind I'd like to punt on the data structure thing--whatever we decide to do I'd like to also refactor proxyconfig.go so that we're consistent and that feels out of scope of this PR.

Reply:
Yeah I don't want to get hung up on this either.